### PR TITLE
M2P-420 Create integration style tests for class OrderManagement

### PR DESCRIPTION
**Pull request by dumega**
# Description
Create integration style tests for class OrderManagement

Fixes: https://boltpay.atlassian.net/browse/M2P-420

#changelog Create integration style tests for class OrderManagement

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [x] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [ ] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# For PR Reviewer 
- [ ] Reviewed unit tests to make sure we are using real components rather than mocks as much as possible?
- [ ] For any major change (observer, new Bolt feature, core Magento interaction) we must add a feature switch, did you verify this?

# Checklist:

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my Jira ticket link and provided a changelog message.

### CODE REVIEW

**Review by danac-gs — APPROVED** [review]
Great work!

File: Test/Unit/Model/Api/OrderManagementTest.php [hunk 1]
```diff
@@ -48,875 +77,1964 @@
...
+                ]
+            ]
+        ];
+        
+        return $requetData;
+    }
+    
+    
+    
+    private function createProduct()
```
Comment (vitaliyreznikov) [inline Test/Unit/Model/Api/OrderManagementTest.php:408]: @johnmccombs1 already created some of helpers. Anyway, we need to move those method out to reuse them in other test classes

**Review by vitaliyreznikov — CHANGES_REQUESTED** [review]
Great job overall!
Just need to move. helpers outside of test class

File: Test/Unit/Model/Api/OrderManagementTest.php (hunk 1 above)
Comment (dumega) [reply Test/Unit/Model/Api/OrderManagementTest.php:401]: I move some functions to class TestUtils and refactor some others to resue the existing util functions.

**Review by danac-gs — APPROVED** [review]

File: Test/Unit/Model/Api/OrderManagementTest.php [hunk 2]
```diff
@@ -48,875 +77,1915 @@
...
-            $mockBuilder->setMethods($methods);
-        } else {
-            $mockBuilder->enableProxyingToOriginalMethods();
-        }
-        $this->currentMock = $mockBuilder->getMock();
+        $addressInfo = $this->getAddressInfo();
+        
+        $quoteId = $quote->getId();
+        
+        $addressData = [
```
Comment (vitaliyreznikov) [inline Test/Unit/Model/Api/OrderManagementTest.php:503]: We have this code few times, could we move it to helper?

File: Test/Unit/Model/Api/OrderManagementTest.php [hunk 3]
```diff
@@ -48,875 +77,1915 @@
...
+            'street'     => $addressInfo['street_address1'],
+            'city'       => $addressInfo['locality'],
+            'email'      => $addressInfo['email_address'],
+            'telephone'  => $addressInfo['phone_number'],
+            'country_id' => $addressInfo['country_code'],
+        ];
+        
+        $orderItems = [];
+        
+        $orderItem = $this->objectManager->create(OrderItem::class);
```
Comment (vitaliyreznikov) [inline Test/Unit/Model/Api/OrderManagementTest.php:518]: helper: getOrderItemByProduct

**Review by vitaliyreznikov — APPROVED** [review]
Huge work! Couple of nits, could you please address them in separate PR?

**Pull request merged**